internal/repository/sync.go

```diff
@@ -64,8 +68,49 @@ func Clone(ctx context.Context, remoteURL, dir, initialBranch string) (*Reposito
 		return nil, errors.Join(ErrCloningRepository, err)
 	}
 
+	sort.Slice(expectedRootKeys, func(i, j int) bool {
+		return expectedRootKeys[i].KeyID < expectedRootKeys[j].KeyID
+	})
+
 	repository := &Repository{r: r}
 
 	slog.Debug("Verifying HEAD...")
+	if len(expectedRootKeys) > 0 {
+		slog.Debug("Getting root keys...")
+		state, err := policy.LoadCurrentState(ctx, r)
+		if err != nil {
+			return repository, errors.Join(ErrCloningRepository, err)
+		}
+
+		rootMetadata, err := state.GetRootMetadata()
+		if err != nil {
+			return repository, errors.Join(ErrCloningRepository, err)
+		}
+
+		if len(rootMetadata.Roles[policy.RootRoleName].KeyIDs) != len(expectedRootKeys) {
```

Comment:
I think we would benefit from having `state.GetRootKeys()` so we have a consistent way of loading root keys for a policy state. The reason is you could look at state.RootMetadata or state.RootPublicKeys. Neither is incorrect as long as we're also checking they match up (I can handle this in #304 where I rework state.Verify).

Reply:
I have added this and a similar function to query all root keys ids